ipatests/test_ipapython/test_ipautil.py

```diff
@@ -508,3 +509,40 @@ def test_run_stderr():
     assert "message" not in str(cm.value)
     assert "message" not in str(cm.value.output)
     assert "message" not in str(cm.value.stderr)
+
+
+@pytest.fixture(scope='function')
+def tcp_listen():
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+        s.bind(('', 0))  # 0 means random port
```

Comment:
Will random port lead to random test failures? We can always accept it as-is and reconsider if we see periodic failures.

Reply:
With port = 0, the OS will use a random, unused, and free port. It won't fail unless all ports > 1024 are exhausted. It's a useful to trick for tests.